test/codecs/CirceDecodersTest.scala

```diff
@@ -141,5 +144,98 @@ class CirceDecodersTest extends WordSpec with MustMatchers {
       checkBankAccountData.accountNumber.value mustBe "12121212"
     }
   }
+
+  "SwitchStateDecoder" should {
+    "decode json" in {
+      decode[SwitchState]("\"On\"") mustBe Right(SwitchState.On)
```

Comment:
Think it's idiomatic to use [`EitherValues`](http://www.scalatest.org/user_guide/using_EitherValues) when testing the value of an `Either`.